packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -70,30 +70,149 @@ export interface DeploymentCircuitBreaker {
 }
 
 /**
- * The ecs metric name
- * Units of this metrics are percentages
+ * The ecs metric class
  */
-export enum EcsMetric {
+export class EcsMetric {
+  /**
+   * Service Connect Metric Map
+   */
+  public static readonly SERVICE_CONNECT_METRIC_MAP: Record<string, string> = {
```

Comment:
What happens if Service Connect adds more metrics and customer needs to add a custom metric before we support them? is there a way we can take another param in the constructor, or a set of props for future compatibility or more complex validation?

Like for example what could you do with this interface in the EcsMetric constructor?
```ts
export interface EcsMetricEnumProps {
  isServiceConnectMetric?: boolean;
  isOtherServiceMetric?: boolean;
}
```